### PR TITLE
CXXTemporaryObjectExpr Support

### LINKED ISSUE #896: [C++ Verification] ERROR: elidable copy/move is not supported in get_expr
```
class Foo {
  public:
    Foo() {};
    ~Foo() {};
    void Execute() {};
};

int main()
{
  auto foo = Foo();
  foo.Execute();
  return 0;
}
```

```
% ~/esbmc/build/src/esbmc/esbmc main2.cpp -I ~/esbmc/src/cpp/library
ESBMC version 7.0.0 64-bit x86_64 macos
Target: 64-bit little-endian x86_64-unknown-macos with esbmclibc
Parsing
main2.cpp:10:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
  auto foo = Foo();
  ^
Converting
ERROR: elidable copy/move is not supported in get_expr
```

### PR DESCRIPTION
**Pull request by kunjsong01**
Patch to fix https://github.com/esbmc/esbmc/issues/896 

Added support for CXXTemporaryObjectExpr

### CODE REVIEW

**Review by lucasccordeiro — APPROVED** [review]
LGTM.

**Comment by lucasccordeiro** [comment]
@kunjsong01: I believe this PR will enable some existing regression tests...

File: src/clang-cpp-frontend/clang_cpp_convert.cpp [hunk 1]
```diff
@@ -737,9 +741,15 @@ bool clang_cpp_convertert::get_constructor_call(
...
   const clang::Decl *objectDecl = it->get<clang::Decl>();
-  if(!objectDecl)
+
+  /*
+   * Distinguish `Foo foo = new foo();` from `Foo foo = foo();`
+   * The latter involves a temporary object and copy constructor call
+   * where the the temporary object construction is considered an argument to
+   * the cpy ctor call's AST node. We shouldn't make a new_object in this case.
+   */
+  if(!objectDecl && constructor_call.arguments().empty())
```
Comment (mikhailramalho) [inline src/clang-cpp-frontend/clang_cpp_convert.cpp]: what happens when the object's constructor has one argument but we construct it using `new`?

**Comment by lucasccordeiro** [comment]
The following tests FAILED:
	1964 - regression/esbmc-cpp11/constructors/Constructor1-6 (Failed)
	1973 - regression/esbmc-cpp11/constructors/Constructor3-1 (Failed)
	1992 - regression/esbmc-cpp/inheritance_bringup/dynamic_cast_simple2 (Failed)
Errors while running CTest
Error: Process completed with exit code 8.

File: src/clang-cpp-frontend/clang_cpp_convert.cpp (hunk 1 above)
Comment (kunjsong01) [reply src/clang-cpp-frontend/clang_cpp_convert.cpp]: failed with not enough arguments for function call: 
https://github.com/esbmc/esbmc/actions/runs/4124493249/jobs/7123810808#step:10:4112 

apparently need to rethink the condition here...

Comment (kunjsong01) [reply src/clang-cpp-frontend/clang_cpp_convert.cpp]: now fixed

**Comment by kunjsong01** [comment]
> The following tests FAILED: 1964 - regression/esbmc-cpp11/constructors/Constructor1-6 (Failed) 1973 - regression/esbmc-cpp11/constructors/Constructor3-1 (Failed) 1992 - regression/esbmc-cpp/inheritance_bringup/dynamic_cast_simple2 (Failed) Errors while running CTest Error: Process completed with exit code 8.

fixed

**Review by lucasccordeiro — APPROVED** [review]
LGTM.

**Review by rafaelsamenezes — APPROVED** [review]

**Pull request merged**